src/nepiada/env/nepiada.py

```diff
@@ -267,22 +279,67 @@ def move_drones(self, actions):
         # Drones that collided with another drone, for reprocessing
         collided_drones = []
 
-        for agent_id in self.agents:
-            agent = self.world.agents[agent_id]
-            action = actions[agent_id]
+        for agent_name in self.agents:
+            agent = self.world.agents[agent_name]
+            action = actions[agent_name]
             status = self.world.grid.move_drone(agent, action)
             if status == -2:
-                collided_drones.append(agent_id)
+                collided_drones.append(agent_name)
             elif status == -1:
                 # Drone collided with boundary
                 pass
 
         # Check collided drones in reverse to see if moving them is possible in this step
-        for agent_id in reversed(collided_drones):
-            agent = self.world.agents[agent_id]
-            action = actions[agent_id]
+        for agent_name in reversed(collided_drones):
+            agent = self.world.agents[agent_name]
+            action = actions[agent_name]
             status = self.world.grid.move_drone(agent, action)
             if status == -2:
                 # Drone collided with another drone
                 pass
 
+    def get_rewards(self):
```

Comment:
Should the rewards be based on the previous position of the agent vs the current position of the agent? i.e the reward is proportional to the action that the agent has just taken?